test/wasm2js/target_js.wast

```diff
@@ -0,0 +1,29 @@
+ (func $0 (param $0 i32) (result i32)
+  (unreachable)
+ )
+ (func $1 (param $0 i32) (param $1 i32)
+  (i32.store
+   (i32.const 1)
+   (i32.xor
+    (i32.shl
```

Comment:
Please add an explanation in the testcase, that even when optimizing we should not turn the xor into something that does not express well in JS.